src/libraries/System.Formats.Tar/src/System/Formats/Tar/SubReadStream.cs

```diff
@@ -147,6 +155,8 @@ protected async ValueTask<int> ReadAsyncCore(Memory<byte> buffer, CancellationTo
         {
             Debug.Assert(!_hasReachedEnd);
 
+            cancellationToken.ThrowIfCancellationRequested();
+
```

Comment:
> I was under the impression that the token cancellation status should be checked on every method

As long as some work was executed between them (so it could have actually changed in the meantime). At least this is how I see it.